File: paddlers-shared-lib/src/api/statistics.rs
use serde::*;

#[derive(Serialize, Deserialize, Debug)]
pub struct FrontendRuntimeStatistics {
    pub browser: BrowserInfo,
    pub session_duration_s: i64,
    pub fps: f64,
}

#[derive(Serialize, Deserialize, Debug)]
pub struct BrowserInfo {
    pub user_agent: String,
    pub inner_width: i32,
    pub inner_height: i32,
    pub outer_width: i32,
    pub outer_height: i32,
}

File: paddlers-frontend/src/logging/statistics.js
export class JsBrowserInfo {
    constructor() {
        this.user_agent = navigator.userAgent;
        this.inner_width = window.inner_width();
        this.inner_height = window.inner_height();
        this.outer_width = window.outer_width();
        this.outer_height = window.outer_height();
    }
}